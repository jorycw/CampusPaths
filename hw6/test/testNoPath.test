# Simple test of breadth-first search algorithm.
# Verifies that it correctly indicates when there is no path found 
# between two nodes. This can be the case when there exists no edges
# between the two nodes or when the direction of the edges don't allow
# for a path from the first node to the second (note for marvel paths
# all paths (edges) will always be two sided)

CreateGraph characters
AddNode characters A
AddNode characters B
AddNode characters C

FindPath characters A C

AddEdge characters A B A2B
AddEdge characters C B C2B

FindPath characters A C

AddNode characters D
AddEdge characters B D B2D

FindPath characters A D
FindPath characters D A
