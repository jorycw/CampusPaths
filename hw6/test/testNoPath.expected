# Simple test of breadth-first search algorithm.
# Verifies that it correctly indicates when there is no path found
# between two nodes. This can be the case when there exists no edges
# between the two nodes or when the direction of the edges don't allow
# for a path from the first node to the second (note for marvel paths
# all paths (edges) will always be two sided)

created graph characters
added node A to characters
added node B to characters
added node C to characters

path from A to C:
no path found

added edge A2B from A to B in characters
added edge C2B from C to B in characters

path from A to C:
no path found

added node D to characters
added edge B2D from B to D in characters

path from A to D:
A to B via A2B
B to D via B2D
path from D to A:
no path found
